baseplate/secrets/store.py

```diff
@@ -0,0 +1,253 @@
+            logger.debug("Loading secrets from %s.", self._path)
+
+            with open(self._path) as f:
+                raw_data = json.load(f)
+
+            self._expiration = datetime.datetime.strptime(
+                raw_data["expiration"], ISO_FORMAT)
+            self._vault_token = raw_data["vault_token"]
+            self._secrets = raw_data["secrets"]
+
+    def make_object_for_context(self, name, server_span):  # pragma: nocover
+        return self
+
+    def get_vault_token(self):
+        """Return a Vault authentication token.
+
+        The token will have policies attached based on the current EC2 server's
+        Vault role. This is only necessary if talking directly to Vault.
+
+        :rtype: :py:class:`str`
+
+        """
+        self._load_if_expired()
+        return self._vault_token
+
+    def get_raw(self, path):
+        """Return a dictionary of key/value pairs for the given secret path.
+
+        This is the raw representation of the secret in the underlying store.
+
+        :rtype: :py:class:`dict`
+
+        """
+        self._load_if_expired()
+
+        try:
+            return self._secrets[path]
+        except KeyError:
+            raise SecretNotFoundError(path)
+
+    def get_simple(self, path):
+        """Decode and return a simple secret.
+
+        Simple secrets are a convention of key/value pairs in the raw secret
+        payload.  The following keys are significant:
+
+        ``type``
+            This must always be ``simple`` for this method.
+        ``value``
+            This contains the raw value of the secret token.
+        ``encoding``
+            (Optional) If present, how to decode the value from how it's
+            encoded at rest (Only ``base64`` currently supported).
+
+        :rtype: :py:class:`bytes`
+
+        """
+        secret_attributes = self.get_raw(path)
+
+        if secret_attributes.get("type") != "simple":
+            raise CorruptSecretError(path, "secret does not have type=simple")
+
+        try:
+            value = secret_attributes["value"]
+        except KeyError:
+            raise CorruptSecretError(path, "secret does not have value")
+
+        encoding = secret_attributes.get("encoding", "identity")
+        return _decode_secret(path, encoding, value)
+
+    def get_versioned(self, path):
+        """Decode and return a versioned secret.
+
+        Versioned secrets are a convention of key/value pairs in the raw secret
+        payload. The following keys are significant:
+
+        ``type``
+            This must always be ``versioned`` for this method.
+        ``current``, ``next``, and ``previous``
+            The raw secret value's versions. ``current`` is the "active"
+            version, which is used for new creation/signing operations.
+            ``previous`` and ``next`` are only used for validation (e.g.
+            checking signatures) to ensure continuity when keys rotate. Both
```

Comment:
Check out how it's used in the message signer for when the previous/next values come into play.